<!-- -->
<launch>
  <rosparam param="/use_sim_time">true</rosparam>
  <arg name="ori_fix" />
  <arg name="person_init_fix" />
  <arg name="person_init_dir" />
  <arg name="device_name" default="/android" />
  <arg name="pf_gyro_sd" default="0.000001" />
  <arg name="pf_sys_noise_sd" default="0.2,0.2,0.087263889,0.00001,0.1" />
  <arg name="pf_gps_xy_bias" default="5" />
  <arg name="pf_perception_xy_sd" default="0.3" />
  <arg name="pf_perception_r_sd" default="0.034905556" />

  <!--<arg name="marker_display_duration" default="0.2"/>                           -->

  <param name="origin_fix" value="$(arg ori_fix)" />

  <node pkg="smartphone_localizer" type="fusion" name="fusion" output="screen">
    <param name="person_initial_fix" value="$(arg person_init_fix)" />
    <param name="person_initial_direction" value="$(arg person_init_dir)" />
    <param name="particlefilter_gyro_standard_deviation" value="$(arg pf_gyro_sd)" />
    <param name="particlefilter_system_noise_standard_deviation" value="$(arg pf_sys_noise_sd)" />
    <param name="particlefitler_gps_xy_bias" value="$(arg pf_gps_xy_bias)" />
    <param name="particlefitler_perception_xy_standard_deviation" value="$(arg pf_perception_xy_sd)" />
    <param name="particlefitler_perception_r_standard_deviation" value="$(arg pf_perception_r_sd)" />
    <remap from="/android/fix" to="$(arg device_name)/fix" />
    <remap from="/android/imu" to="$(arg device_name)/imu" />
  </node>

  <node pkg="smartphone_localizer" type="phonesensors_raw" name="phonesensors_raw">
    <param name="origin_fix" value="$(arg ori_fix)" />
    <remap from="/android/fix" to="$(arg device_name)/fix" />
    <remap from="/android/imu" to="$(arg device_name)/imu" />
    <param name="person_initial_direction" value="$(arg person_init_dir)" />
    <param name="person_frame_id" value="person_sensor_raw" />
  </node>


  <!--<node pkg="tf2_ros" type="static_transform_publisher" name="base_link2rslidar" args="0. 0. 0. -3.1415 0. 0. /base_link /rslidar" required="true"/>-->

  <!--<node pkg="detected_objects_visualizer" type="visualize_detected_objects" name="cluster_detect_visualization_01"
          output="screen" ns="/detection/lidar_detector" >
    <param name="marker_display_duration"            value="$(arg marker_display_duration)" />
  </node>                     -->

</launch>
